proto/internal/impl/v1/impl.proto

```diff
@@ -77,6 +77,7 @@ message RelationMetadata {
   }
 
   RelationKind kind = 1;
+  repeated string type_annotations = 2;
```

Comment:
Put this under a TypeAnnotations message so we can change it in the future if we want